src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -38,10 +40,30 @@ public function onKernelRequest(GetResponseEvent $event)
         $request->setSession($session);
     }
 
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        if (!$session = $event->getRequest()->getSession()) {
+            return;
+        }
+
+        if ($session->isStarted() || ($session instanceof Session && ((\method_exists(Session::class, 'hasBeenStarted') && $session->hasBeenStarted()) || (\method_exists(Session::class, 'isEmpty') && !$session->isEmpty())))) {
```

Comment:
Why does it need to maek the response uncachable when the session has been started but there is no data?

Reply:
Because this is the semantic of the cache control header in HTTP: if the session is used, whatever it contains, then it means the page can be different for another user, and thus the page cannot be cached.
That's not new at all.

Reply:
Hum, your question might be related to the isEmpty call on this line.
In fact this call is only required for maximum compatibility with soon-legacy versions of HttpKernel, where hasBeenStarted is not available.
We could also bump the minimum version requirement from 3.3 to 3.4.4.
I'll do that :)

Reply:
Hum, your question might be related to the isEmpty call on this line.
In fact this call is only required for maximum compatibility with soon-legacy versions of HttpKernel, where hasBeenStarted is not available.
We could also bump the minimum version requirement from 3.3 to 3.4.4.
I'll do that :)